mono/utils/mono-membar.h

```diff
@@ -77,7 +77,23 @@ static inline void mono_memory_write_barrier (void)
 	mono_memory_barrier ();
 }
 #else
-#error "Don't know how to do memory barriers!"
+
+static inline void mono_memory_barrier (void)
+{
+	g_assert(0 && "This function is not yet implemented for the Unity platform.");
+}
+
+static inline void mono_memory_read_barrier (void)
+{
+	g_assert(0 && "This function is not yet implemented for the Unity platform.");
+}
+
+static inline void mono_memory_write_barrier (void)
+{
+	g_assert(0 && "This function is not yet implemented for the Unity platform.");
+}
+
+//#error "Don't know how to do memory barriers!"
```

Comment:
What is the proper approach here? Should we leave these stubs in and put them behind the `UNITY_PLATFORM` define?

Reply:
Hmm, i do prefer compile/link errors in case like this. Probably fine now but we should forward to Unity impl or error longer term.

Reply:
Ok, I'll surround this with the `UNITY_PLATFORM` define, and put the `#error` back in for other cases.